### PR TITLE
make enabliong statsd configurable

### PR DESCRIPTION
**Pull request by GingerMoon**
after this pr, I'd like to add prometheus support

### CODE REVIEW

**Comment by GingerMoon** [comment]
@spikelu2016 please feel free to let me know if # ave

**Comment by spikelu2016** [comment]
Can we make this backward compatible with people having `StatsProvider` set? This change is a breaking change.

File: cmd/bricksllm/.env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CONFIG_FILE_NAME=
+CONFIG_FILE_NAME=./config_local.json
```
Comment (spikelu2016) [inline cmd/bricksllm/.env:1]: when you set this, the default application behavior is using variables from config_local.json. can we set this to empty so we don't change the default behavior?

**Comment by spikelu2016** [comment]
if `StatsEnabled` is set to false, we will have `nil pointer dereference` error. can you help ensure that even when `StatsEnabled` is set to false, this error does not happen?

File: cmd/bricksllm/.env (hunk 1 above)
Comment (GingerMoon) [reply cmd/bricksllm/.env]: Done.

**Comment by GingerMoon** [comment]
Thanks for pointing this. I will add an interface and makes statsd the default implementation.

**Comment by GingerMoon** [comment]
Done.

**Pull request merged**